### PR TITLE
Move `namelist.update` and `model.update_namelist_pio` into `Step`

### PR DESCRIPTION
**Pull request by xylar**
These are more intuitive if they're methods of Step instead. This merge also updates the test cases and core framework that is affected.  It also updates the documentation with these changes.

### CODE REVIEW

**Comment by xylar** [comment]
@matthewhoffman, I'm mostly look for a review of the documentation changes from you.  Let me know if this would have helped you with the confusion you were having.

File: compass/step.py [hunk 1]
```diff
@@ -340,6 +341,70 @@ def add_namelist_options(self, options, out_name=None, mode='forward'):
 
         namelist_list.append(dict(options=options, mode=mode))
 
+    def update_namelist_at_runtime(self, options, out_name=None):
+        """
+        Update an existing namelist file with additional options.  This would
+        typically be used for namelist options that are only known at runtime,
+        not during setup.  For example, the number of PIO tasks and the stride
```
Comment (matthewhoffman) [inline compass/step.py:348]: Should this sentence be removed now with the new function you've added specifically for PIO?

File: docs/developers_guide/organization.rst [hunk 2]
```diff
@@ -1604,12 +1604,34 @@ an option passed in when creating this step.
   Namelist values must be of type ``str``, so use ``'{}'.format(value)`` to
   convert a numerical value to a string.
 
+
+.. _dev_step_update_namelist_options:
```
Comment (matthewhoffman) [inline docs/developers_guide/organization.rst:1608]: I like that you've moved this namelist topic to live with the other namelist info.

**Review by matthewhoffman — APPROVED** [review]
I left one minor comment about something you may want to fix in a docstring.  I like your change to the documentation itself.

By the way, there is a missing : on line 1518 of docs/developers_guide/organization.rst.  I tried to change it myself with a suggestion, but I think one can only make a suggestion for lines that were touched by the PR.

File: compass/step.py [hunk 3]
```diff
@@ -340,6 +341,70 @@ def add_namelist_options(self, options, out_name=None, mode='forward'):
...
+        between tasks, which are related to the number of nodes and cores.
```
Comment (xylar) [inline compass/step.py:349]:
```suggestion
        not during setup, typically those related to the number of nodes and cores.
```

File: compass/step.py (hunk 1 above)
Comment (xylar) [reply compass/step.py]: Good suggestion.

**Pull request merged**